blog/vscode-remote-development-setup-and-usage.mdx

```diff
@@ -0,0 +1,260 @@
+on-the-go the same way I do on my desktop, without having to unplug anything
+(beside the laptop’s charger). And once I’m back, I can reuse my home setup.
+
+Finally, having such a setup with two machines is also an advantage for
+development. Two machines means potentially two OSs, and it can become handy to
+test some code on two completely different, yet very “real” environments. In my
+case, My desktop runs on Windows and my laptop on Ubuntu, meaning I can easily
+test OS-specific features if the need arises (it used to be really useful for
+Internet Explorer, luckily gone are those days! 😅), while keeping the Linux CLI
+environment I prefer.
+
+### Cons
+
+Here’s a catch though: two machines, also means two _running_ machines, which in
+turn means more energy consumption as well. This is definitely not an
+environment friendly method !
+
+During specific use cases, I sometimes encounter issues with port forwarding
+that force me to get back to my laptop’s keyboard, for example when using
+[Docker](https://www.docker.com/) with [Traefik](https://traefik.io/traefik/),
+VS Code is sometimes confused about what port to forward and where to forward it
+to.
+
+On the hardware side, this method renders your laptop (or host) a bit less
+useful, as most of the time you’ll work at home you won’t touch it much, besides
+turning it on or off. We will see in the next part other methods I tried that
+actually uses both machines actively.
+
+On the management side, depending on how you manage your local machine, and how
+(if) you use it outside of work, you’ll have to take extra care of not mixing
+work and personal things, which can be extra tricky for some, especially if
+you’re pretty new to working at home.
+
+## Comparison with other methods I have tried
+
+### Why not use a dock ?
+
+Most docks are really great at what they are meant to do. Unfortunately, this
+doesn’t include sharing peripherals between devices, which was my goal here. I
+still use my dock, however in my current setup, since I use the same keyboard /
+mouse for both my machines, it would mean plugging / unplugging those everyday
+which is kind of painful.
+
+### Using both machine efficiently: Synergy / Barrier
+
+This was by far my most preferred method. [Synergy](https://symless.com/synergy)
+(and its open source clone [Barrier](https://github.com/debauchee/barrier)) is a
+keyboard and mouse sharing software, that enable a single pair of peripherals to
+work on many different machines at the same time. Unlike VS Code remote (or SSH)
+only, using Synergy you are able to actually use every device’s UI as well, and
+coupled to VS Code Remote, you can basically use “any” number of machines the
+same way you’ll be using one.
+
+Unfortunately, I had a bug on Ubuntu using Barrier that made me stop using this
+method some times ago as I couldn’t find a solution. If this issue was to be
+fixed however, I would hop in the Barrier train any day !
```

Comment:
**nitpick:** there is no space before punctuation

**suggestion:** do a full search/replace on the blog post ` !` -> `!` (same for ` :` maybe)
```suggestion
fixed however, I would hop in the Barrier train any day!
```